src/main/kotlin/de/bund/digitalservice/useid/identification/MockDatasource.kt

```diff
@@ -1,23 +1,45 @@
 package de.bund.digitalservice.useid.identification
 
+import mu.KotlinLogging
 import org.springframework.stereotype.Repository
 import reactor.core.publisher.Mono
 import java.util.UUID
 
 @Repository
 class MockDatasource {
     private val sessions = mutableListOf<IdentificationSession>()
+    private val log = KotlinLogging.logger {}
 
-    fun save(tcTokenUrl: String, refreshAddress: String, requestAttributes: List<String>): Mono<IdentificationSession> {
-        val currentSessionId = UUID.randomUUID()
+    fun save(refreshAddress: String, requestAttributes: List<String>): Mono<IdentificationSession> {
+        val useIDSessionId = UUID.randomUUID()
 
-        val identificationSession = IdentificationSession(refreshAddress, requestAttributes, currentSessionId, tcTokenUrl)
+        val identificationSession = IdentificationSession(
+            refreshAddress = refreshAddress,
+            requestAttributes = requestAttributes,
+            useIDSessionId = useIDSessionId
+        )
         sessions.add(identificationSession)
 
         return Mono.just(identificationSession)
     }
 
-    fun sessionExists(sessionId: UUID): Boolean {
-        return sessions.any { it.sessionId == sessionId }
+    fun sessionExists(useIDSessionId: UUID): Boolean {
+        return sessions.any { it.useIDSessionId == useIDSessionId }
+    }
+
+    fun findByIdOrFail(useIDSessionId: UUID): Mono<IdentificationSession> {
+        if (!sessionExists(useIDSessionId)) {
+            throw Error("no session found with useIDSessionId $useIDSessionId")
+        }
+        return Mono.just(sessions.find { it.useIDSessionId == useIDSessionId }!!)
+    }
+
+    fun updateEIDSessionId(useIDSessionId: UUID, eIDSessionId: UUID) {
+        // use !! here bc findByIdOrFail was already executed
```

Comment:
We shouldn't do such a thing, because this method could be used in the future somewhere else or in  a different way and this wouldn't hold. So either pass in the session, which was received by `findByIdOrFail` or call `findByIdOrFail` again.

Reply:
+1 strongly agree! A method should never depend on another method to be executed before.

Reply:
And it does not hurt to call `findByIdOrFail` again here.

Reply:
+1